### ISSUE TITLE
Run "headful" tests inside Xvfb

### ISSUE THREAD

**Issue by atroche** [atroche/rust-headless-chrome#37]
It's pretty annoying to have Chrome windows pop up and steal focus when you're running the test suite. I've messed around with running this project with Xvfb before (for the purposes of video recording), and it's not hard to work with.

The only problem is that it means anyone who wants to run the tests would need to install Xvfb.

Thoughts @r-darwish @lukaslueg @messense ?

**Comment by lukaslueg** [comment]
Why do they execute with `headless(false)` anyway?

**Comment by atroche** [comment]
In non-headless mode, if you try to start multiple processes, they'll share the same Chrome instance unless they're different user profiles. I guess it's not such an important test now that we definitely do use a different profile each time.

Maybe I could just kill the test instead?

I've got a hunch that there will be subtle differences between headful and headless mode that will need to be tested, but that's really just a hunch, and it probably isn't worth bringing in Xvfb as a dependency.

**Comment by atroche** [comment]
Interestingly, one of my integration tests (the one that involved a Dropbox file picker pop up) just *would not* work in headless mode, even though it worked fine headfully. I spent a long time trying to debug that, and still couldn't work out why.

**Comment by lukaslueg** [comment]
Everything should Just Work with `headless`, as otherwise the lib can't be used where there are no screens (think CI). It to cumbersome and hacky for people to have xvfb in their dependency tree and, by extension, in their package manager's installation list (again think CI)

**Comment by atroche** [comment]
Agreed! But I will try to get a minimal example of that popup weirdness I experienced in headless mode, just to satisfy my own curiosity :)

**Comment by r-darwish** [comment]
I don't think that tests should be responsible for running themselves in xvfb. It will also make testing in macOS and Windows impossible. We should run as many test as possible in headless mode in my opinion.

**Comment by lukaslueg** [comment]
Let me re-open: There seem to be things which only work in non-headless. While I do think that tests should always execute in headless (unless `#[ignored]`), there could be a code-example in `/examples` on how to do a xvfb-head.

Also see https://github.com/atroche/rust-headless-chrome/pull/42#discussion_r257505188
